### ISSUE TITLE
[crash] kotlinx.coroutines.CompletionHandlerException at kotlinx.coroutines.JobSupport.completeStateFinalization(JobSupport.kt:15)

### ISSUE THREAD

**Issue by ebalazs-sv** [mozilla-mobile/fenix#16397]
### Precondition
Log in with an account.
## Steps to reproduce
1. Launch Fenix and go to three-dot menu -> Settings;
2. Tap the email address from the Account section;
3. Tap "Sign out";
### Expected behavior
The user is signed out from the account and the Settings page is displayed.
### Actual behavior
Fenix crashes: https://crash-stats.mozilla.org/report/index/82793f28-3326-4d0e-9bb7-c7d570201105#tab-details
This is not reproducible all the time, 1 crash out of 5 tries.

`kotlinx.coroutines.CompletionHandlerException
	at kotlinx.coroutines.JobSupport.completeStateFinalization(JobSupport.kt:15)
	at kotlinx.coroutines.JobSupport.finalizeFinishingState(JobSupport.kt:20)
	at kotlinx.coroutines.JobSupport.tryMakeCompleting(JobSupport.kt:26)
	at kotlinx.coroutines.JobSupport.makeCompletingOnce$kotlinx_coroutines_core(JobSupport.kt:2)
	at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:1)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:9)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:13)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:216)
	at android.app.ActivityThread.main(ActivityThread.java:7266)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:975)`

### Device information

* Android device: Samsung Galaxy Tab S3 (Android 8);
* Fenix version: Nightly 201105 05:01 (Build #2015773801) GV 84 from 11/5
 
### Notes
Also, the Disconnect button is invisible: https://github.com/mozilla-mobile/fenix/issues/16398

**Comment by AndiAJ** [comment]
Reproducible on **`84.0.0-beta.3`** using a OnePlus A3 (Android 6.0.1)

1971ba37-1de6-4007-b851-8dff5306881d
kotlinx.coroutines.CompletionHandlerException: Exception in completion handler InvokeOnCompletion[InvokeOnCompletion@5613965] for StandaloneCoroutine{Cancelled}@300663a
* Sentry: https://sentry.prod.mozaws.net/operations/fenix-fennec-beta/?query=133e6f2cbc664ee1b39855e05d32a86f
* Socorro: https://crash-stats.mozilla.org/report/index/bp-e398dfaa-f04c-42db-8468-eb2860201202
----
kotlinx.coroutines.CompletionHandlerException: Exception in completion handler InvokeOnCompletion[InvokeOnCompletion@5613965] for StandaloneCoroutine{Cancelled}@300663a
at kotlinx.coroutines.JobSupport.completeStateFinalization(JobSupport.kt:15)
at kotlinx.coroutines.JobSupport.finalizeFinishingState(JobSupport.kt:20)
at kotlinx.coroutines.JobSupport.tryMakeCompleting(JobSupport.kt:26)
at kotlinx.coroutines.JobSupport.makeCompletingOnce$kotlinx_coroutines_core(JobSupport.kt:2)
at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:1)
at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:9)
at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:13)
at android.os.Handler.handleCallback(Handler.java:739)
at android.os.Handler.dispatchMessage(Handler.java:95)
at android.os.Looper.loop(Looper.java:152)
at android.app.ActivityThread.main(ActivityThread.java:5507)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
Caused by: java.lang.IllegalStateException: Fragment SignOutFragment{1fb58b9} (fa242aed-2729-4d65-9b98-afbcce1e1b89)} not associated with a fragment manager.
at androidx.fragment.app.Fragment.getParentFragmentManager(Fragment.java:2)
at androidx.navigation.fragment.NavHostFragment.findNavController(NavHostFragment.java:3)
at androidx.core.app.AppOpsManagerCompat.findNavController(AppOpsManagerCompat.java:2)
at org.mozilla.fenix.settings.account.SignOutFragment$onViewCreated$1$2.invoke(SignOutFragment.kt:2)
at kotlinx.coroutines.InvokeOnCompletion.invoke(JobSupport.kt:4)
at kotlinx.coroutines.JobSupport.completeStateFinalization(JobSupport.kt:13)
... 13 more

**Comment by softvision-miralobontiu** [comment]
Reproducible also on Samsung Galaxy Tab s6 (Android 10), on the latest beta 84.0.0-beta.3:

0397c3c4-61e9-4b58-820f-e8deaafa74cc
kotlinx.coroutines.CompletionHandlerException: Exception in completion handler InvokeOnCompletion[InvokeOnCompletion@254b30c] for StandaloneCoroutine{Cancelled}@3e61455
----
kotlinx.coroutines.CompletionHandlerException: Exception in completion handler InvokeOnCompletion[InvokeOnCompletion@254b30c] for StandaloneCoroutine{Cancelled}@3e61455
at kotlinx.coroutines.JobSupport.completeStateFinalization(JobSupport.kt:15)
at kotlinx.coroutines.JobSupport.finalizeFinishingState(JobSupport.kt:20)
at kotlinx.coroutines.JobSupport.tryMakeCompleting(JobSupport.kt:26)
at kotlinx.coroutines.JobSupport.makeCompletingOnce$kotlinx_coroutines_core(JobSupport.kt:2)
at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:1)
at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:9)
at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:13)
at android.os.Handler.handleCallback(Handler.java:883)
at android.os.Handler.dispatchMessage(Handler.java:100)
at android.os.Looper.loop(Looper.java:237)
at android.app.ActivityThread.main(ActivityThread.java:8167)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
Caused by: java.lang.IllegalStateException: Fragment SignOutFragment{ba5ffaf} (b6cc80ec-be3e-4425-9089-799685641ae1)} not associated with a fragment manager.
at androidx.fragment.app.Fragment.getParentFragmentManager(Fragment.java:2)
at androidx.navigation.fragment.NavHostFragment.findNavController(NavHostFragment.java:3)
at androidx.core.app.AppOpsManagerCompat.findNavController(AppOpsManagerCompat.java:2)
at org.mozilla.fenix.settings.account.SignOutFragment$onViewCreated$1$2.invoke(SignOutFragment.kt:2)
at kotlinx.coroutines.InvokeOnCompletion.invoke(JobSupport.kt:4)
at kotlinx.coroutines.JobSupport.completeStateFinalization(JobSupport.kt:13)
... 13 more

**Comment by eliserichards** [comment]
Looks like we might need to manually associate it with a fragment manager: 
```Caused by: java.lang.IllegalStateException: Fragment SignOutFragment{ba5ffaf} (b6cc80ec-be3e-4425-9089-799685641ae1)} not associated with a fragment manager.
at androidx.fragment.app.Fragment.getParentFragmentManager(Fragment.java:2)
at androidx.navigation.fragment.NavHostFragment.findNavController(NavHostFragment.java:3)
at androidx.core.app.AppOpsManagerCompat.findNavController(AppOpsManagerCompat.java:2)
at org.mozilla.fenix.settings.account.SignOutFragment$onViewCreated$1$2.invoke(SignOutFragment.kt:2)
```

**Comment by ekager** [comment]
I have a potential fix in https://github.com/mozilla-mobile/fenix/pull/16823 but I can't repro on my emulator. Could someone who could repro pull down this PR to see if it fixes it?

**Comment by ekager** [comment]
Just want to make sure ya'll saw this sorry for the tag! If someone who can reliably reproduce could test the APK from the above speculative fix that would be great! https://firefox-ci-tc.services.mozilla.com/tasks/dHcq4siESvSXcGPsSSehUQ

@softvision-miralobontiu @ebalazs-sv @AndiAJ

**Comment by ebalazs-sv** [comment]
@ekager Hi! We had tested this on the given build with the following devices:
- OnePlus A3 (Android 6.0.1);
- Samsung Galaxy S9 (Android 8);
- OnePlus 6T (Android 9);
- Pixel 2 (Android 9);
- Google Pixel (Android 10);
- Google Pixel 3a (Android 11);

and the app does not crash when signing out.
Also the Disconnect button is not invisible with this fix.

**Comment by ekager** [comment]
Thank you so much! :)

**Comment by ebalazs-sv** [comment]
Verified as fixed on both the latest Nightly 201208 and 84.0.0-beta.4 using the following devices:
- Google Pixel 4 XL (Android 11);
- Samsung Galaxy Tab S6 (Android 10);
- Google Pixel 3 XL (Android 9)
- Motorola Moto G6 (Android 8);
- OnePlus A3 (Android 6.0.1).

I will close this issue.
